src/Core/Grid/Search/TemporarySearchCriteria.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Search;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class TemporarySearchCriteria is temporary search criteria class. Should be removed.
+ */
+class TemporarySearchCriteria implements SearchCriteriaInterface
+{
+    private $request;
+
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    public function getOrderBy()
+    {
+        return $this->request->get('orderBy', 'id_log');
```

Comment:
Maybe use `->request->query->get` instead of directly `get` even if it's a temporary class.

Reply:
What is the interest? Temporary means really temporary in this specific case, this class will be removed by https://github.com/PrestaShop/PrestaShop/pull/8984

Reply:
Huhu, yeah in this case, it's really really temporay ^^